src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -104,11 +102,6 @@ public GeneratedFileDirectoryInfo(DocumentId documentId, Type generatorType)
             // The IVsRunningDocumentTable is a free-threaded VS service that allows fetching of the service and advising events
             // to be done without implicitly marshalling to the UI thread.
             _runningDocumentTable = _serviceProvider.GetService<SVsRunningDocumentTable, IVsRunningDocumentTable>();
-            _runningDocumentTableEventTracker = new RunningDocumentTableEventTracker(
```

Comment:
Can the constructor call for some reason do something useful?

Reply:
@jasonmalinowski?

Reply:
Yes, this constructor call, as a side effect, subscribes to events.

Reply:
And the integration test failures are the source generator tests, so that's probably because of this.

Reply:
@jasonmalinowski These event listeners are not allowed to be subscribed within the constructor of a MEF component.

Reply:
The shell has a specific implementation that specifically allows this use, for precisely this reason.